### DISCUSSION TITLE
Diablo 1 Installation Troubleshooting

### DISCUSSION THREAD

**Discussion by ChrisVapo** [diasurgical/DevilutionX#1625] [Q&A] [answered]
Hello all! 👋

I’ve recently purchased Diablo 1 and am currently in the process of trying to figure out to get this to work on MacOS. I was able to install Homebrew and have tried to follow the requirements to the best of my ability.

I am hung up on the step “Extracting the .MPQs from the GoG installer” 

Now on to the part where the Terminal retrieves info about the GOG Diablo Installer: 

For some reason, I get an error upon running the following command in the Terminal: “innoextract ––info 'setup_diablo_1.09_hellfire_v2_(30038).exe”

The error states that it could not open the file. 

Surely there must have been something I’ve overlooked. Any help would be appreciated guys. Thanks in advance!

**Comment by AJenbo** [comment]
what is the exact error?

what does innoextract --version show?

**Reply by ChrisVapo** [reply]
Sorry, here is the additional info you asked for: innoextract 1.9
Extracts installers created by Inno Setup 1.2.10 to 6.0.5

**Comment by ChrisVapo** [comment]
Hello,

The error message is given as follows: Could not  open file
“setup_diablo_1.09_hellfire_v2_(30038).exe”
Done with 1 error.

I receive this error when trying to obtain info about the GOG Diablo
Installer.

**Reply by AJenbo** [reply]
Ok, that seems to be either because you are giving it the wrong location of your file (what is it) or you do not have the right permissions.

**Comment by ChrisVapo** [comment, marked as answer]
I don’t think it’s a permissions issue. I can read and write as far as
privileges go.

However, looking at where this Installer is located, it says: Macintosh HD
> Users > christophersanchez > Downloads

**Reply by AJenbo** [reply]
Sounds like you are not giving it the correct path of the file, the simplest solution is to copy it to the same folder as you are running the commands from. "Macintosh HD" isn't a path, but the label of one of your partitions.

**Reply by ChrisVapo** [reply]
Would this be considered the file path?

https://www.gog.com/account
https://cdn-hw.gog.com/secure/offline/1412601690/1412601690/52336385680515400/5291/setup_diablo_1.09_hellfire_v2_%2830038%29.exe?ttl=1618903650&hw_l=106&hw_token=cc758138db634ecc64ca574d887e495d&_token=eyJhbGciOiJBMTI4S1ciLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0.sRG2Sib-DblZzY6lWrSwNFvAFJF0BsrWQugU5-CR6s1L81HwLG1DUQ.Uvw1sh_IG0lKyXw7nolCGQ.4dKrCTy0HbCJaeQFhhPO0qLrqM1WV-0BxdE816_OKIUJWLmlBd7HnGMmsNVclzmBVjVK-QkRcuOIR1q0qeXDCoQky69C_1vYu-hkcaVvcsWq88mjNZy9YodgZezg3Xp7IaT2KhDpsZfYpPbMz3IwKSoNY5Vb-uaoIYrI-vyg_zZ53parCQ5eSNneWo0Ce3HnCZxaETAgDuYAYT13hqSs7TO2oVEKAg-F5b4HCF1UKn-eJZShg7WEOxBax0LqVeBoOALNbctNtPh8bygPbRLHlrUp2ZdFEp1AZRVlB4j3xJthPNMgUa7JEN-YLbeWZ0Qol8UET7g2gmpNL-1hI80nz2_pmyNAtcg1rp_CjmAzvAJ58vF0LXNruiF7EP13bD0YJx7ORWIgVGSyMBS5tixHpnQzDnpid72ZRDnvSPc4mm5U_9D-8aWlk3x-GljkzXbCMJRq-095vvZHuDE6urTAi6yAYIsHhwQ7CqQfZmYduM4hgwORdjyKoRyIGZe6cWII.vJtkKXmMtAVV3JaAar20uQ&source=hw

**Reply by ChrisVapo** [reply]
Okay, I think I was able to get further into it. This is what I see now:

Usage: innoextract [options] <setup file(s)>

Extract files from an Inno Setup installer.
For multi-part installers only specify the exe file.

Generic options:
  -h [ --help ]                 Show supported options
  -v [ --version ]              Print version information
  --license                     Show license information

Actions:
  -t [ --test ]                 Only verify checksums, don't write anything
  -e [ --extract ]              Extract files (default action)
  -l [ --list ]                 Only list files, don't write anything
  --list-sizes                  List file sizes
  --list-checksums              List file checksums
  -i [ --info ]                 Print information about the installer
  --list-languages              List languages supported by the installer
  --gog-game-id                 Determine the installer's GOG.com game ID
  --show-password               Show password check information
  --check-password              Abort if the password is incorrect
  -V [ --data-version ]         Only print the data version

Modifiers:
  --codepage arg                Encoding for ANSI strings
  --collisions arg              How to handle duplicate files
  --default-language arg        Default language for renaming
  --dump                        Dump contents without converting filenames
  -L [ --lowercase ]            Convert extracted filenames to lower-case
  -T [ --timestamps ] arg       Timezone for file times or "local" or "none"
  -d [ --output-dir ] arg       Extract files into the given directory
  -P [ --password ] arg         Password for encrypted files
  --password-file arg           File to load password from
  -g [ --gog ]                  Extract additional archives from GOG.com 
                                installers
  --no-gog-galaxy               Don't re-assemble GOG Galaxy file parts
  -n [ --no-extract-unknown ]   Don't extract unknown Inno Setup versions

Filters:
  -m [ --exclude-temp ]         Don't extract temporary files
  --language arg                Extract only files for this language
  --language-only               Only extract language-specific files
  -I [ --include ] arg          Extract only files that match this path

Display options:
  -q [ --quiet ]                Output less information
  -s [ --silent ]               Output only error/warning information
  --no-warn-unused              Don't warn on unused .bin files
  -c [ --color ] [=arg(=1)]     Enable/disable color output
  -p [ --progress ] [=arg(=1)]  Enable/disable the progress bar

Extracts installers created by Inno Setup 1.2.10 to 6.0.5

innoextract 1.9 (C) 2011-2020 Daniel Scharrer <daniel@constexpr.org>
This is free software with absolutely no warranty.
christophersanchez@Christophers-iMac cmake %

**Reply by AJenbo** [reply]
> Would this be considered the file path?
> 
> https://www.gog.com/account

No, that is an internet URL. And it requires your account.

Right-click the file and select file info:
![image](https://user-images.githubusercontent.com/204594/115422472-3735c280-a1fd-11eb-9741-db62997e3515.png)

**Reply by AJenbo** [reply]
> Okay, I think I was able to get further into it. This is what I see now:
> 
> Usage: innoextract [options] <setup file(s)>

Sadly no, this is the help text for innoextract because it saw you using it in a non-sensical way.

**Reply by ChrisVapo** [reply]
<img width="390" alt="Screen Shot 2021-04-20 at 10 29 07 AM" src="https://user-images.githubusercontent.com/82861515/115423808-befedb00-a1c3-11eb-895d-f90058dacb28.png">


Uploading a screenshot. This is what I see.

**Reply by AJenbo** [reply]
It looks like you put it in your Downloads folder, so you path is:
`/Users/christophersanchez/Downloads/setup_diablo_1.09_hellfire_v2_(30038).exe`
or `~/Downloads/setup_diablo_1.09_hellfire_v2_(30038).exe` for short.

**Reply by ChrisVapo** [reply]
<img width="647" alt="Screen Shot 2021-04-20 at 10 43 15 AM" src="https://user-images.githubusercontent.com/82861515/115425507-4c8efa80-a1c5-11eb-8d85-02eab14d9059.png">

Thank you. How do I rename this appropriately?  Do I still include "innoextract --info" and paste the rest of the info or?..

**Reply by AJenbo** [reply]
You have two options:

A: Move setup_diablo_1.09_hellfire_v2_(30038).exe to the folder where you are running the commands from (you have omitted this so I don't know).

B: Whenever the guide says `setup_diablo_1.09_hellfire_v2_(30038).exe`, change it to `/Users/christophersanchez/Downloads/setup_diablo_1.09_hellfire_v2_(30038).exe`

**Reply by ChrisVapo** [reply]
I'm not sure how to go about with Route A. I'm simply inputting these commands through "Terminal". I'm uncertain if there is a folder for it. 

I also tried Option B and was given the same error as before. 
<img width="567" alt="Screen Shot 2021-04-20 at 11 08 16 AM" src="https://user-images.githubusercontent.com/82861515/115429158-c2e12c00-a1c8-11eb-844e-d587b0b5b190.png">

**Reply by AJenbo** [reply]
Good news, your latest screenshot includes your current location. You are in the home folder.

If you move the file from the Download folder to the Home folder you should be able to use the commands as they appear in the guide.

https://www.cnet.com/how-to/how-to-find-your-macs-home-folder-and-add-it-to-finder/

**Reply by ChrisVapo** [reply]
I was able to get into the home folder. I placed the home location on the sidebar "christophersanchez". I also got the GOG Installer from "Downloads", clicked and then dragged the installer to the "christophersanchez" folder. I hope I'm on the right track.
<img width="1152" alt="Screen Shot 2021-04-20 at 12 45 32 PM" src="https://user-images.githubusercontent.com/82861515/115441880-004cb600-a1d7-11eb-83b4-0eabd0205ef2.png">

**Reply by ChrisVapo** [reply]
I believe some progress has been made. 
<img width="570" alt="Screen Shot 2021-04-20 at 12 53 36 PM" src="https://user-images.githubusercontent.com/82861515/115442358-879a2980-a1d7-11eb-8457-721ea375ff86.png">

**Reply by AJenbo** [reply]
> I was able to get into the home folder. I placed the home location on the sidebar "christophersanchez". I also got the GOG Installer from "Downloads", clicked and then dragged the installer to the "christophersanchez" folder. I hope I'm on the right track.
> <img alt="Screen Shot 2021-04-20 at 12 45 32 PM" width="1152" src="https://user-images.githubusercontent.com/82861515/115441880-004cb600-a1d7-11eb-83b4-0eabd0205ef2.png">

The file in your screenshot clearly is called `GOG_Galaxy_Diablo_Hellfire.exe` and not `setup_diablo_1.09_hellfire_v2_(30038).exe`, it's also only ~500KB so not the right one as that should be about 800MB.

**Reply by AJenbo** [reply]
> I believe some progress has been made.
> <img alt="Screen Shot 2021-04-20 at 12 53 36 PM" width="570" src="https://user-images.githubusercontent.com/82861515/115442358-879a2980-a1d7-11eb-8457-721ea375ff86.png">

Yes, it appears you have correctly done option B. Now you just need to do the last step:
https://github.com/diasurgical/devilutionX/wiki/Extracting-the-.MPQs-from-the-GoG-installer#extract-diabdatmpq-and-the-four-hellfire-mpqs-from-the-installer

**Reply by ChrisVapo** [reply]
Okay. I was able to successfully extract the MPQs. I see that they are now within my Home Folder. In my case, what is the specified location that these files need to go to?

<img width="820" alt="Screen Shot 2021-04-20 at 2 23 29 PM" src="https://user-images.githubusercontent.com/82861515/115453331-224d3500-a1e5-11eb-880d-446750f8206a.png">

**Reply by qndel** [reply]
Please read the readme

**Reply by AJenbo** [reply]
Put the MPQ files next to devilutionx, or : https://github.com/diasurgical/devilutionX/blob/master/docs/installing.md#installing

**Reply by ChrisVapo** [reply]
Thanks @AJenbo! I finally got it to work. I was curious. Is there a way to do this exact process for Diablo 2?

**Reply by AJenbo** [reply]
Yes, but it will probably take you multiple years to do so. Joke aside, we have no interest in working on Diablo 2. You will need a Mac running macOS 10.14 or older to play Diablo 2.
